pkg/cloudprovider/providers/gce/gce_clusters.go

```diff
@@ -41,24 +44,30 @@ func (gce *GCECloud) ListClusters(ctx context.Context) ([]string, error) {
 }
 
 func (gce *GCECloud) GetManagedClusters(ctx context.Context) ([]*container.Cluster, error) {
-	managedClusters := []*container.Cluster{}
-	for _, zone := range gce.managedZones {
-		clusters, err := gce.getClustersInZone(zone)
-		if err != nil {
-			return nil, err
-		}
-		managedClusters = append(managedClusters, clusters...)
+	var location string
+	if len(gce.managedZones) > 1 {
+		// Multiple managed zones means this is a regional cluster
+		// so use the regional location and not the zone.
+		location = gce.region
+	} else if len(gce.managedZones) == 1 {
+		location = gce.managedZones[0]
+	} else {
+		return nil, errors.New(fmt.Sprintf("no zones associated with this cluster(%s)", gce.ProjectID()))
+	}
+	clusters, err := gce.getClustersInLocation(location)
+	if err != nil {
+		return nil, err
 	}
 
-	return managedClusters, nil
+	return clusters, nil
 }
 
 func (gce *GCECloud) Master(ctx context.Context, clusterName string) (string, error) {
 	return "k8s-" + clusterName + "-master.internal", nil
 }
 
 func (gce *GCECloud) listClustersInZone(zone string) ([]string, error) {
```

Comment:
This function is not public so hopefully you can fix the name and the parameter name?

Reply:
oh wait, I misunderstood. nvm.